crates/flux-infer/src/infer.rs

```diff
@@ -475,6 +481,12 @@ impl Sub {
                 infcx.unify_exprs(&path1.to_expr(), &path2.to_expr());
                 self.tys(infcx, &ty1, ty2)
             }
+            (TyKind::StrgRef(_, path1, ty1), TyKind::StrgRef(_, path2, ty2)) => {
+                env.unfold_strg_ref(infcx, path1, ty1)?;
```

Comment:
I assume not doing anything didn't work because when processing the ensures after the call we use `update_path`, so we would need something like `update_or_insert_path`. 

Maybe we can leave a comment explaining that. Or more importantly, we are assuming functions always give back the "ownership" of the location so even though we should technically "consume" the ownership and remove the location from the environment, the type is always going to be overwritten. (there's a check for this btw, if you write an &strg we require an ensures for that location for the signature to be well-formed)